core/templates/dev/head/pages/exploration_editor/translation_tab/TranslationStatusService.js

```diff
@@ -57,21 +121,24 @@ oppia.factory('TranslationStatusService', [
             INTERACTION_SPECS[interactionId].is_terminal) {
             allContentId = ['content'];
           }
-          explorationAudioRequiredCount += allContentId.length;
+          explorationContentRequiredCount += allContentId.length;
           allContentId.forEach(function(contentId) {
-            var availableTranslationLanguageCodes = (
-              contentIdsToAudioTranslations.getAudioLanguageCodes(contentId));
-            if (availableTranslationLanguageCodes.indexOf(langCode) > -1) {
-              var audioTranslation = contentIdsToAudioTranslations
-                .getAudioTranslation(contentId, langCode);
-              if (audioTranslation.needsUpdate) {
-                stateNeedsUpdateWarnings[stateName] = NEEDS_UPDATE_MESSAGE;
-              }
-            } else {
+            var availabilityStatus = _getContentAvailabilityStatus(
+              stateName, contentId);
+            if (!availabilityStatus[0]) {
```

Comment:
These will be more readable if the availabilityStatus was an object with descriptive keys.